framework/src/main/java/org/checkerframework/framework/type/TypeFromExpressionVisitor.java

```diff
@@ -270,7 +270,22 @@ public AnnotatedTypeMirror visitMemberSelect(MemberSelectTree tree, AnnotatedTyp
         } else {
             // tree must be a field access, so get the type of the expression, and then call
             // asMemberOf.
-            AnnotatedTypeMirror t = f.getAnnotatedType(tree.getExpression());
+            AnnotatedTypeMirror t;
+            if (f instanceof GenericAnnotatedTypeFactory) {
+                // If calling GenericAnnotatedTypeFactory#getAnnotatedTypeLhs(Tree lhsTree) to
+                // get the type of this MemberSelectTree, flow refinement is disabled. However,
+                // we want the receiver to have the refined type because type
+                // systems can use receiver-dependent qualifiers for viewpoint adaptation.
+                // Thus, we re-enable the flow refinement for a while just for the receiver
+                // expression.
+                // See framework/tests/viewpointtest/TestgetAnnotatedLHS.java:26 for a concrete
```

Comment:
Also update the file name here. I would remove the line number, in particular as it is already out of date.

Reply:
Umm, why did you resolve this? Line 26 points to the second line of the comment in the test. Either point to the first line or just the whole file.